AnnService/src/Helper/AsyncFileReader.cpp

```diff
@@ -93,33 +92,15 @@ namespace SPTAG {
 
                     int fileid = (readRequest->m_status >> 16);
                     if (fileid != currFileId) {
-                        (handlers[currFileId].get()->*f)(readRequests + currReqStart, i - currReqStart);
+                        handlers[currFileId]->BatchReadFile(readRequests + currReqStart, i - currReqStart);
                         currFileId = fileid;
                         currReqStart = i;
                     }
                 }
                 if (currReqStart < num) {
-                    (handlers[currFileId].get()->*f)(readRequests + currReqStart, num - currReqStart);
-                }
-            }
-        }
-
-        void BatchReadFileAsync(std::vector<std::shared_ptr<Helper::DiskIO>>& handlers, AsyncReadRequest* readRequests, int num)
-        {
-            
-            CallOnAppropriateBatch(handlers, readRequests, num, &DiskIO::BatchReadFile);
-
-            for (int i = 0; i < num; i++) {
-                AsyncReadRequest* readRequest = &(readRequests[i]);
-                
-                if (readRequest->m_success && readRequest->m_callback)
-                {
-                    readRequest->m_callback(true);
-                    readRequest->m_callback = nullptr;
+                    handlers[currFileId]->BatchReadFile(readRequests + currReqStart, num - currReqStart);
                 }
             }
-
-            CallOnAppropriateBatch(handlers, readRequests, num, &DiskIO::BatchCleanRequests);
```

Comment:
We still need to clean requests in windows build, to release the refcounted buffers in internal implementation

Reply:
I have modified the DiskIO part so that let the diskio to clean the context instead of our library.